Fix leaks in WebKitTestRunner

ddkilzer opened the pull request:
#### 2c5c5725a6c0231437df23060a1954f5b2f87e8c
<pre>
Fix leaks in WebKitTestRunner
<a href="https://bugs.webkit.org/show_bug.cgi?id=243711">https://bugs.webkit.org/show_bug.cgi?id=243711</a>
&lt;rdar://98357151&gt;

Reviewed by Geoffrey Garen.

* Source/WebKit/UIProcess/API/C/WKDownloadRef.cpp:
(WKDownloadSetClient):
- Fix leak of API::String since
  TestController::decideDestinationWithSuggestedFilename()
  returns a +1 retained WKStringRef.
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::runOpenPanel):
- Fix leak of WKArrayRef since
  WKOpenPanelParametersCopyAllowedMIMETypes() returns a +1
  retained WKArrayRef.
* Tools/WebKitTestRunner/mac/main.mm:
(disableAppNapInUIProcess):
- Use idomatic NeverDestroyed&lt;&gt; declaration to make `leaks`
  stop reporting a false positive leak.

Canonical link: <a href="https://commits.webkit.org/253271@main">https://commits.webkit.org/253271@main</a>
</pre>

geoffreygaren commented on Tools/WebKitTestRunner/mac/main.mm:
```diff
@@ -53,12 +53,15 @@ static void setDefaultsToConsistentValuesForTesting()
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSActivityOptions options = (NSActivityUserInitiatedAllowingIdleSystemSleep | NSActivityLatencyCritical) & ~(NSActivitySuddenTerminationDisabled | NSActivityAutomaticTerminationDisabled);
+        assertion.get() = retainPtr([[NSProcessInfo processInfo] beginActivityWithOptions:options reason:@"WebKitTestRunner should not be subject to process suppression"]);
```
Is the call to retainPtr() required here? 'assertion' is declared with a concrete type, so I don't think we need help with template argument deduction.

ddkilzer replied:
It is not strictly necessary.  I was thinking we did this explicitly in most cases, but I think it's only when we want to use `auto` for a new variable declaration so we don't have to type out the whole `RetainPtr<>` type.

I'll remove it.

geoffreygaren reviewed (approved):
r=me

Commit:
Fix leaks in WebKitTestRunner
https://bugs.webkit.org/show_bug.cgi?id=243711
<rdar://98357151>

Reviewed by Geoffrey Garen.

* Source/WebKit/UIProcess/API/C/WKDownloadRef.cpp:
(WKDownloadSetClient):
- Fix leak of API::String since
  TestController::decideDestinationWithSuggestedFilename()
  returns a +1 retained WKStringRef.
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::runOpenPanel):
- Fix leak of WKArrayRef since
  WKOpenPanelParametersCopyAllowedMIMETypes() returns a +1
  retained WKArrayRef.
* Tools/WebKitTestRunner/mac/main.mm:
(disableAppNapInUIProcess):
- Use idomatic NeverDestroyed<> declaration to make `leaks`
  stop reporting a false positive leak.

Canonical link: https://commits.webkit.org/253271@main
